react-paw-mailmerge/src/Helpers/CsvFunctions.tsx

```diff
@@ -15,35 +24,59 @@ export function ParseCsv(myFile:File, options:CsvOptions, resolve: (s: Array<obj
         dynamicTyping: true, // dynamically type, so that numbers are numbers and bool are bools and not strings
         keepEmptyRows: false, 
         complete: function(results:ParseResult<object>, file:File) { // This is what is called when we successfully parse
-            resolve(NormalizeJsonObjectResult(results, options.dataHasHeader));
+            resolve(NormalizeJsonObjectResult(results, options));
         },
         error:  function(results:object, file:File) { // This is what is called when we fail to parse
             console.error("Errors:", results, file);
             reject("There was an error", results);
         },
-        header: options.dataHasHeader,
+        header: options.header,
         skipEmptyLines: true
     } as ParseLocalConfig<unknown, File>;
 
     parse(myFile, config);
   }
 
-function NormalizeJsonObjectResult(results:ParseResult<object>, hasHeader:boolean) : Array<object>
+function HandleNullValues(val: any, nullFieldOption : NullFieldOptionEnum) : any
```

Comment:
There might be an easier way to do this. The config for the parser accepts a `transform` function that is applied on every value. Using `transform` will also simplify parts of `NormalizeJsonObjectResult`

Reply:
I'm not convinced that I'm normalizing this in the best way. My hunch is that I'm going to eventually want an array of arrays of key value pairs to iterate over. When I create the buttons, I'm probably going to revisit this output and will look into transform at that juncture. Thank you for calling it out!